python/tank/util/session.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Session management for Toolkit.
+"""
+
+import os
+
+from tank_vendor.shotgun_api3 import Shotgun
+from tank_vendor.shotgun_api3 import AuthenticationFault
+from ConfigParser import SafeConfigParser
+from .shotgun import get_associated_sg_config_data
+
+# Configure logging
+import logging
+logger = logging.getLogger("sgtk-session")
+# logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+
+def _get_cached_login_info_location(base_url):
+    """
+    Returns the location of the session file on disk for a specific site.
+    :param base_url: The site we want the login information for.
+    :returns: A dictionary with keys "login" and "session_token".
+    """
+    from tank.api import get_site_cache_root
+    return os.path.join(
+        get_site_cache_root(base_url),
+        "login.ini"
+    )
+
+
+def is_session_token_cached():
+    """
+    Returns if there is a cached session token for the current user.
+    :returns: True is there is, False otherwise.
+    """
+    if get_cached_login_info(get_associated_sg_config_data()["host"]):
+        return True
+    else:
+
+        return False
+
+
+def get_cached_login_info(base_url):
+    """
+    Returns the cached login info if found.
+    :returns: Returns a dictionnary with keys hostname, login and session_token or None
+    """
+    # Retrieve the location of the cached info
+    info_path = _get_cached_login_info_location(base_url)
+    # Nothing was cached, return an empty dictionary.
+    if not os.path.exists(info_path):
+        logger.debug("No cache found at %s", info_path)
+        return {}
```

Comment:
Yeah, I fixed this locally when I started writing unit tests. :)

From: Manne Öhrström <notifications@github.com<mailto:notifications@github.com>>
Reply-To: shotgunsoftware/tk-core <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, February 17, 2015 at 9:32 AM
To: shotgunsoftware/tk-core <tk-core@noreply.github.com<mailto:tk-core@noreply.github.com>>
Cc: JF Boismenu <jean.francois.boismenu@autodesk.com<mailto:jean.francois.boismenu@autodesk.com>>
Subject: Re: [tk-core] For #28030, [security] User/password sign in on tank command with session caching (#129)


In python/tank/util/session.py<https://github.com/shotgunsoftware/tk-core/pull/129#discussion_r24818711>:

> +    else:
> +
> +        return False
> +
> +
> +def get_cached_login_info(base_url):
> +    """
> +    Returns the cached login info if found.
> +    :returns: Returns a dictionnary with keys hostname, login and session_token or None
> +    """
> +    # Retrieve the location of the cached info
> +    info_path = _get_cached_login_info_location(base_url)
> +    # Nothing was cached, return an empty dictionary.
> +    if not os.path.exists(info_path):
> +        logger.debug("No cache found at %s", info_path)
> +        return {}


maybe returning None is better? (that's what the doscstrings says)

—
Reply to this email directly or view it on GitHub<https://github.com/shotgunsoftware/tk-core/pull/129/files#r24818711>.